scripts/irods/test/test_igroupadmin.py

```diff
@@ -72,6 +71,61 @@ def tearDownClass(self):
+                with session.make_session_for_existing_user(username, passwd, lib.get_hostname(), self.admin.zone_name) as user_session:
+                    user_session.assert_icommand('icd')
+            finally:
+                self.admin.run_icommand(['iadmin','rmuser',username])
+
+    def test_remove_from_group_scenarios__issue_6888(self):
+        """Thoroughly test atg, rfg (remove from group) and transfer of responsiblity to another group administrator"""
+        rodsuser_name = 'alice'
+        try:
+            # Use the groupadmin account to create group and set itself up as administrator.
+            self.groupadmin.assert_icommand(['igroupadmin', 'mkgroup', self.group])
+            self.groupadmin.assert_icommand(['igroupadmin', 'atg', self.group, self.groupadmin.username])
+
+            # Another groupadmin is needed for this test.
+            self.groupadmin2 = session.mkuser_and_return_session('groupadmin', 'groupadmin2', 'rods', lib.get_hostname())
+
+            # Test another user (a rodsuser) can be created and added to the group by this.
+            self.groupadmin.assert_icommand(['igroupadmin', 'mkuser', rodsuser_name])
+            self.groupadmin.assert_icommand(['igroupadmin', 'atg', self.group, rodsuser_name])
+
+            user_with_zone = lambda user_name,zn='': '{}#{}'.format(user_name, zn if zn else self.groupadmin.zone_name)
+
+            # Test that the groupadmin itself, and the new rodsuser, are members of the group.
+            self.groupadmin.assert_icommand(['igroupadmin', 'lg', self.group, ], 'STDOUT_MULTILINE', [user_with_zone(rodsuser_name),
+                                                                                                      user_with_zone(self.groupadmin.username)])
+
+            # Test removal of users, and self, from a group, as well as assignment of a new admin for the group via "atg".
+            self.groupadmin.assert_icommand(['igroupadmin', 'rfg', self.group, rodsuser_name])         # remove rodsuser
+            self.groupadmin.assert_icommand(['igroupadmin', 'atg', self.group, self.groupadmin2.username])  # add other groupadmin
+            self.groupadmin.assert_icommand(['igroupadmin', 'rfg', self.group, self.groupadmin.username])   # remove self
+
+            # Test that groupadmin2 is now a member of the group, and that groupadmin and the rodsuser have successfully been removed.
+            rc, out, _ = self.groupadmin2.assert_icommand(['igroupadmin', 'lg', self.group], 'STDOUT_MULTILINE', user_with_zone(self.groupadmin2.username))
+            self.assertEqual(rc, 0)
+            self.assertNotIn(user_with_zone(rodsuser_name), out)
+            self.assertNotIn(user_with_zone(self.groupadmin.username), out)
+
+            # Test that the new groupadmin can do some necessary things.
+            self.groupadmin2.assert_icommand(['igroupadmin', 'atg', self.group, rodsuser_name], desired_rc = 0)
```

Comment:
Should we assert that `rodsuser_name` is a member of the group after this `atg` call?

Reply:
perhaps